### ISSUE TITLE
False-positive unneeded-not when comparing dict views

### ISSUE THREAD

**Issue by natedogith1** [pylint-dev/pylint#3668]
dict-view comparison triggers unneeded-not, despite the comparison being set-like, and therefore the "not {}.items() <= {}.items()" to "{}.items() > {}.items()" being a functionality change.

Simple example that reproduces the issue
```python
def func():
    if not {}.items() <= {}.items():
        pass
```

pylint --version
pylint 2.4.4
astroid 2.3.3
Python 3.7.3+ (heads/3.7:e64d21b187, Apr 23 2019, 10:57:08)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]

**Comment by natedogith1** [comment]
```python
>>> a = {"a": 1}
>>> b = {"b": 2}
>>> not a.items() <= b.items()
True
>>> a.items() > b.items()
False
```
